tests/utils/series-utils-tests.js

```diff
@@ -0,0 +1,143 @@
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import test from 'tape';
+import React from 'react';
+import {mount} from 'enzyme';
+
+import {
+  isSeriesChild,
+  getSeriesPropsFromChildren,
+  getSeriesChildren,
+  getStackedData
+} from 'utils/series-utils';
+import LineSeries from 'plot/series/line-series';
+import XAxis from 'plot/axis/x-axis';
+import HorizontalBarSeries from 'plot/series/vertical-rect-series';
+
+test('series-utils #isSeriesChild', t => {
+  const series = React.createElement(LineSeries, {data: []});
+  t.ok(isSeriesChild(series), 'Should return true for series');
+  const axis = React.createElement(XAxis, {
+    xRange: [0, 1],
+    xDomain: [0, 1],
+    xType: 'linear',
+    width: 100,
+    height: 100,
+    top: 0,
+    left: 0
+  });
+  t.notOk(isSeriesChild(axis), 'Should return false for non-series');
+  t.end();
+});
+
+test('series-utils #getSeriesChildren', t => {
+  const children = [
+    (<span key="wild"> This squid is heavy </span>),
+    (<input key="wacky" placeholder="i wish i hadnt bought it" />),
+    (<LineSeries key="woah" data={[]} />)
+  ];
+  const $ = mount(
+    <div width={300} height={300}>
+      {children}
+    </div>
+  );
+  const expectedChildren = [{...children[2], key: '.$woah'}];
+  t.deepEqual(getSeriesChildren($.props().children), expectedChildren, 'should find the correct children');
+  t.end();
+});
+
+const arePropsValid = seriesProps => {
+  return typeof seriesProps._colorValue !== 'undefined' &&
+    typeof seriesProps._opacityValue !== 'undefined' &&
+    typeof seriesProps.ref === 'string' &&
+    typeof seriesProps.sameTypeIndex === 'number' &&
+    typeof seriesProps.sameTypeTotal === 'number' &&
+    typeof seriesProps.seriesIndex === 'number';
+};
+
+test('series-utils #collectSeriesTypesInfo', t => {
+  const result = getSeriesPropsFromChildren([
+    React.createElement(LineSeries, {data: []}),
+    React.createElement(LineSeries, {data: []})
+  ]);
+  t.ok(result.length === 2, 'Returns array of proper size');
+  result.forEach((props, i) => t.ok(arePropsValid(props), `Props #${i} are valid`));
+  t.end();
+});
+
+test('series-utils #getStackedData', t => {
+  let children = [
```

Comment:
I actually kinda disagree, i think data processing tests should be on the verbose side. It's important to be able to read them if you are unsure about how they work. On the flip side i think they should all be kept together for test-output legibility